### PR TITLE
Richb CSS (and other) Tweaks

### PR DESCRIPTION
**Pull request by richb-hanover**
This commit is good enough to criticize. 

Big question: 

- I made a bunch of changes to dir-index.cgi - it didn't seem to generate valid files otherwise. Have I missed something? 

Big-ish changes:

- I added a README.md to tell what this repo is, and how to test it
- `test.sh` rebuilds each of the files that it generates in production to be sure they look OK
- Updated SampleData to use current (OpenWrt 21.02.0) resources
- Removed the @media queries - they didn't help, and made the page look odd at various browser widths. The only downside is that on very wide screens, the first column gets too wide. We can fix if that's a problem.

Cosmetic:

- Smaller header logo (4em)
- Link color matches logo color (it looked odd to use the same color as the links in the OpenWrt wiki)

### CODE REVIEW

**Comment by aparcar** [comment]
I think a relative path to the css file breaks it for subfolders, e.g. targets/

**Comment by richb-hanover** [comment]
I will look at that tomorrow. ENOTIME today. Thanks

**Comment by aparcar** [comment]
Please don't do this PR again master but again the `little-steps` branch

**Comment by richb-hanover** [comment]
Yikes! You're right. I'll send a new PR

File: dir-index.cgi [hunk 1]
```diff
@@ -197,48 +110,129 @@ sub printh1 {
...
-  printh1($virt);
-  print "<hr>";
+
+  # HTML heading
+  print "<!DOCTYPE html>\n";
+  print "<html lang='en'>\n";
+  print "<head>\n";
+  print "<meta charset='utf-8'/>\n";
+  print "<link rel='stylesheet' href='./static/style.css' />\n";
+  printf "<title>Index of %s</title>\n", $virt;
```
Comment (jow-) [inline dir-index.cgi:162]: Make sure to HTML escape `$virt`

File: dir-index.cgi [hunk 2]
```diff
@@ -197,48 +110,129 @@ sub printh1 {
...
+
+}
+
+# printheader - print the banner at the top of the page
+#
+sub printheader{
+    print <<EOT;
+      <header>
+      <div>
+        <img src="./static/logo.png" alt="Download OpenWrt Firmware">
```
Comment (jow-) [inline dir-index.cgi:209]: Maybe we should consider embedding the logo as SVG directly into the CSS file. One less resource file to worry about.

File: dir-index.cgi [hunk 3]
```diff
@@ -197,48 +110,129 @@ sub printh1 {
...
+
+}
+
+# printfooter - a place to put everything else on the page
+# Should remain within the 'container' div
+#
+sub printfooter {
+print <<EOT;
+  <footer>
+  </footer>\n
```
Comment (jow-) [inline dir-index.cgi:222]: Maybe simply add a further blank line instead of the `\n` escape.

**Review by jow- — COMMENTED** [review]
A couple of small remarks below, nothing critical

**Pull request closed without merging**